### ISSUE TITLE
disabled state not announced for link in Firefox.

### ISSUE THREAD

**Issue by maxability** [nvaccess/nvda#6886]
## When HTML5 disabled attribute is used for an anchor element, NVDA is not recognizing  the state on Mozilla firefox.
<a href="uri" disabled />something</a>

**Comment by feerrenrut** [comment]
Is it valid to have a disabled attribute on an anchor element? Best answer to this I could find was on [W3Schools](http://www.w3schools.com/tags/att_disabled.asp) which says that it is not, however there maybe something newer that I should look at. What should the disabled attribute do on a link?

**Comment by maxability** [comment]
Similar to aria-disabled="true" that announces Unavailable along with link text disabled attribute should also announce unavailable when user navigates to the element. Providing focus to disabled anchor elements might be overwriting browser behavior. I am still not sure if it is a good practice to provide focus to disabled elements or not. It might be different from case to case and developer should be able to judge, in any of the cases NVDA announcing unavailable for elements that contains disabled attribute is required.

**Comment by jcsteh** [comment]
Disabled is not a valid attribute on the A tag. See [the A tag on MDN](https://developer.mozilla.org/en/docs/Web/HTML/Element/a). Note that it isn't a global attribute, either.

**Comment by michelhebert** [comment]
Form elements are the only elements that use the "disabled" attribute. 

I'd like to add that if a developer needs a focusable, yet non-editable form elements, they should be using the "readonly" attribute. This will allow the element to be focused, and read by the screen reader, but not editable.

W3C ref: https://www.w3.org/TR/html5/disabled-elements.html

**Comment by maxability** [comment]
I got it, thank you. I somehow find use of HTML5 disabled and aria-disabled elements used differently. Even the specs are different. aria-disabled can be used globally while disabled should be used only for form elements. NVDA exactly works as per the spec. Thanks for correcting me.
